### ISSUE TITLE
Broken type assertion with structs embedding

### ISSUE THREAD

**Issue by damz** [gopherjs/gopherjs#757]
The `$assertType` helper only considers the method set when asserting if a type implement an interface. But with struct embedding, it is possible for a struct field to shadow a method.

This fails `gopherjs run`:

```
package main

type Doer interface {
	Do()
}

type Embedded struct{}

func (e *Embedded) Do() {}

type A struct {
	Embedded
	Do string
}

func main() {
	var a interface{}
	a = &A{}
	if aa, ok := a.(Doer); ok {
		aa.Do()
	}
}
```

with:

```
TypeError: aa.Do is not a function
```

**Comment by theclapp** [comment]
If anyone else was confused (as I was :), the point is not that aa.Do() should call A.Embedded.Do(), the point is that the type assertion should've failed: `a` is not actually a Doer:

```go
        // In the regular Go playground
	if aa, ok := a.(Doer); ok {
		aa.Do()
	} else {
		println("not a doer")
	}
// => not a doer
```

**Comment by damz** [comment]
@theclapp That's correct, I updated to the description in an effort to make that clearer.

**Comment by damz** [comment]
So, is `$assertType` wrong, and needs to look into properties of the type too, or should this method not be in the methodset of the type to begin with?

**Comment by grantnelson-wf** [comment]
This should be fixed in master now

**Comment by nevkontakte** [comment]
Closing as resolved, thanks for the fix!
